File: support/src/main/java/me/chriskyle/library/support/rxview/java/RxView.java
package me.chriskyle.library.support.rxview.java;

import android.support.annotation.CheckResult;
import android.support.annotation.NonNull;
import android.view.View;

import io.reactivex.Observable;

import static me.chriskyle.library.support.rxview.java.internal.Preconditions.checkNotNull;

public final class RxView {

    @CheckResult
    @NonNull
    public static Observable<Object> clicks(View view) {
        checkNotNull(view, "view == null");
        return new ViewClickObservable(view);
    }

    private RxView() {
        throw new AssertionError("No instances.");
    }
}

File: support/src/main/java/me/chriskyle/library/support/rxview/RxView.kt
@file:Suppress(
        names = ["NOTHING_TO_INLINE"]
)

package me.chriskyle.library.support.rxview

import android.view.View
import io.reactivex.Observable
import me.chriskyle.library.support.rxview.internal.VoidToUnit
import me.chriskyle.library.support.rxview.java.RxView

inline fun View.clicks(): Observable<Unit> = RxView.clicks(this).map(VoidToUnit)

